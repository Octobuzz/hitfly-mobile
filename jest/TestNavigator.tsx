// https://github.com/react-navigation/rfcs/issues/75
import React, { Children } from 'react'
import {
  createAppContainer,
  createSwitchNavigator,
  NavigationParams,
} from 'react-navigation'

const TestNavigator = ({
  children,
  params,
}: {
  children: any
  params?: NavigationParams
}) => {
  const Navigator = createAppContainer(
    createSwitchNavigator({
      TestScreen: { screen: () => Children.only(children), params },
    }),
  )
  return <Navigator />
}

export default TestNavigator
