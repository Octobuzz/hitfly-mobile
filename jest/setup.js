import 'jest-styled-components'
import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock'

jest.mock('react-native-vector-icons/Foundation', () => 'Icon')
jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon')
jest.mock('react-native-vector-icons/Feather', () => 'Icon')
jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon')
jest.mock('react-native-vector-icons/SimpleLineIcons', () => 'Icon')

jest.mock('@react-native-community/async-storage', () => mockAsyncStorage)

jest.mock('react-native-track-player', () => {
  return {
    addEventListener: jest.fn(),
    registerEventHandler: jest.fn(),
    registerPlaybackService: jest.fn(),
    setupPlayer: jest.fn(),
    destroy: jest.fn(),
    updateOptions: jest.fn(),
    add: jest.fn(),
    remove: jest.fn(),
    skip: jest.fn(),
    skipToNext: jest.fn(),
    skipToPrevious: jest.fn(),
    removeUpcomingTracks: jest.fn(),
    // playback commands
    reset: jest.fn(),
    play: jest.fn(),
    pause: jest.fn(),
    stop: jest.fn(),
    seekTo: jest.fn(),
    setVolume: jest.fn(),
    setRate: jest.fn(),
    // player getters
    getQueue: jest.fn(),
    getTrack: jest.fn(),
    getCurrentTrack: jest.fn(),
    getVolume: jest.fn(),
    getDuration: jest.fn(),
    getPosition: jest.fn(),
    getBufferedPosition: jest.fn(),
    getState: jest.fn(),
    getRate: jest.fn(),
  }
})

jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native/Libraries/Components/View/View')
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  }
})
