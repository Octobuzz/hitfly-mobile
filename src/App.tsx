import React from 'react'
import './reactotron.config'
import { YellowBox } from 'react-native'
import { ThemeProvider } from 'styled-components'
import ApolloClient, { InMemoryCache } from 'apollo-boost'
import { persistCache } from 'apollo-cache-persist'
import { ApolloProvider } from '@apollo/react-hooks'
import AppNavigator from './navigation'
import { storageInstance, storage } from './utils'
import theme from './theme'
import { names, storageKeys } from './constants'

// либы до сих пор используют это
YellowBox.ignoreWarnings([
  'Warning: componentWillMount is deprecated',
  'Warning: componentWillUpdate is deprecated',
  'Warning: componentWillReceiveProps is deprecated',
])

async function createApolloClient(): Promise<ApolloClient<InMemoryCache>> {
  const cache = new InMemoryCache({})

  await persistCache({
    cache,
    // @ts-ignore
    storage: storageInstance,
  })

  const token = await storage.getItem(storageKeys.AUTH_TOKEN)

  const client = new ApolloClient<InMemoryCache>({
    cache,
    headers: {
      'X-TOKEN-AUTH': token,
    },
    uri: names.BASE_URL,
  })

  return client
}

class App extends React.Component<{}, { isReady: boolean }> {
  client?: ApolloClient<InMemoryCache>
  state = {
    isReady: false,
  }

  async componentDidMount() {
    this.client = await createApolloClient()
    this.setState({ isReady: true })
  }

  render() {
    const { isReady } = this.state
    return (
      isReady &&
      this.client && (
        <ApolloProvider client={this.client}>
          <ThemeProvider theme={theme}>
            <AppNavigator />
          </ThemeProvider>
        </ApolloProvider>
      )
    )
  }
}

export default App
