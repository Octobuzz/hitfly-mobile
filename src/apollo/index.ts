import * as _storage from './storage'
export const storage = _storage

export * from './queries'
export * from './schemas'
export * from './mutations'
export * from './commonTypes'

export { default as setupClient } from './setupClient'
