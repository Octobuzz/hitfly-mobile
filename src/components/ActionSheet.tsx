import React, { forwardRef, Ref } from 'react'
import { StyleSheet } from 'react-native'
import {
  ActionSheetCustom,
  ActionSheetCustomProps,
} from 'react-native-actionsheet'
import { getBottomSpace } from 'react-native-iphone-x-helper'
import theme from 'src/theme'

interface Props extends ActionSheetCustomProps {
  // TS говно, не захотел подхватить ref
  ref: Ref<ActionSheetCustom>
}

export type ActionSheetInstance = ActionSheetCustom

const ActionSheet: React.FC<Props> = forwardRef<ActionSheetCustom, Props>(
  (props, ref) => {
    return (
      <ActionSheetCustom
        {...props}
        buttonUnderlayColor={theme.colors.buttonUnderlayColor}
        tintColor={theme.colors.white}
        ref={ref}
        styles={styles}
      />
    )
  },
)

// Для библиотеки так
const styles = StyleSheet.create({
  wrapper: { paddingBottom: getBottomSpace() },
  body: {
    backgroundColor: theme.colors.black,
  },
  messageText: {
    color: theme.colors.white,
    fontSize: 14,
    textAlign: 'center',
  },
  messageBox: {
    height: 50,
    paddingBottom: 0,
    backgroundColor: theme.colors.black,
  },
  buttonBox: {
    height: 50,
    backgroundColor: theme.colors.black,
  },
  buttonText: {
    color: theme.colors.white,
  },
  cancelButtonBox: {
    height: 50,
    marginTop: 6,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.colors.black,
  },
})

export default ActionSheet
