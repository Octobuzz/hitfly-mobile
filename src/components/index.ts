export * from './views'
export * from './fields'
export * from './buttons'

export { default as TextBase } from './TextBase'
export { default as GenreItem } from './GenreItem'
export { default as TextWithLines } from './TextWithLines'
