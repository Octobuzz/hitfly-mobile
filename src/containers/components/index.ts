export * from './SocialAuth'

export { default as SocialAuth } from './SocialAuth'
export { default as HeaderRightButtons } from './HeaderRightButtons'
