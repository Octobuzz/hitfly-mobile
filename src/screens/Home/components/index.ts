export { default as StarsSection } from './StarsSection'
export { default as TracksSection } from './TracksSection'
export { default as SectionHeader } from './SectionHeader'
export { default as SectionWrapper } from './SectionWrapper'
export { default as PlaylistHeader } from './PlaylistHeader'
export { default as CollectionSection } from './CollectionSection'
