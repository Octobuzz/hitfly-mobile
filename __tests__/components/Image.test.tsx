import React from 'react'
import { Image, DarkenImage, SvgImage } from 'src/components'
import { render } from '../../jest/test-utils'

describe('SvgImage', () => {
  it('renders correctly with default_track', () => {
    const uri = 'http://somerul.com/default_track.svg'
    const { asJSON } = render(<SvgImage uri={uri} />)
    expect(asJSON()).toMatchSnapshot()
  })
  it('renders correctly with default_profile', () => {
    const uri = 'http://somerul.com/default_profile.svg'
    const { asJSON } = render(<SvgImage uri={uri} />)
    expect(asJSON()).toMatchSnapshot()
  })
  it('renders correctly with default_album', () => {
    const uri = 'http://somerul.com/default_album.svg'
    const { asJSON } = render(<SvgImage uri={uri} />)
    expect(asJSON()).toMatchSnapshot()
  })
  it('renders correctly with default_musicgroup', () => {
    const uri = 'http://somerul.com/default_musicgroup.svg'
    const { asJSON } = render(<SvgImage uri={uri} />)
    expect(asJSON()).toMatchSnapshot()
  })
  it('renders correctly with default_playlist', () => {
    const uri = 'http://somerul.com/default_playlist.svg'
    const { asJSON } = render(<SvgImage uri={uri} />)
    expect(asJSON()).toMatchSnapshot()
  })
})

describe('Image', () => {
  it('renders correctly with number source', () => {
    const { asJSON } = render(<Image source={1} />)
    expect(asJSON()).toMatchSnapshot()
  })

  it('renders correctly with not svg source', () => {
    const { asJSON } = render(<Image source={{ uri: 'not svg' }} />)
    expect(asJSON()).toMatchSnapshot()
  })
})

describe('DarkenImage', () => {
  it('renders correctly', () => {
    const { asJSON } = render(<DarkenImage source={1} />)
    expect(asJSON()).toMatchSnapshot()
  })
})
