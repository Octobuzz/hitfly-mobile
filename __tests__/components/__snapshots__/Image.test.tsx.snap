// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkenImage renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={null}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(49, 49, 49, 0.1)",
            "height": "100%",
            "position": "absolute",
            "width": "100%",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`Image renders correctly with not svg source 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={
        Object {
          "uri": "not svg",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Image renders correctly with number source 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={null}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`SvgImage renders correctly with default_album 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/default_images/album.png",
      }
    }
  />
</View>
`;

exports[`SvgImage renders correctly with default_musicgroup 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/default_images/group.png",
      }
    }
  />
</View>
`;

exports[`SvgImage renders correctly with default_playlist 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/default_images/group.png",
      }
    }
  />
</View>
`;

exports[`SvgImage renders correctly with default_profile 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/default_images/profile.png",
      }
    }
  />
</View>
`;

exports[`SvgImage renders correctly with default_track 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/default_images/music.png",
      }
    }
  />
</View>
`;
